src/Npgsql/MultiplexingConnectorPool.cs

```diff
@@ -261,15 +260,24 @@ bool WriteCommand(NpgsqlConnector connector, NpgsqlCommand command, ref Multiple
                 // to buffer in memory), and the actual flush will occur at the level above. For cases where the
                 // command overflows the buffer, async I/O is done, and we schedule continuations separately -
                 // but the main thread continues to handle other commands on other connectors.
+                var isFullyPrepared = false;
+
                 if (_autoPrepare)
                 {
                     // TODO: Need to log based on numPrepared like in non-multiplexing mode...
-                    var numPrepared = 0;
+                    isFullyPrepared = true;
                     for (var i = 0; i < command.InternalBatchCommands.Count; i++)
-                        if (command.InternalBatchCommands[i].TryAutoPrepare(connector))
-                            numPrepared++;
+                        if (!command.InternalBatchCommands[i].TryAutoPrepare(connector))
+                            isFullyPrepared = false;
                 }
 
+                if (Log.IsEnabled(NpgsqlLogLevel.Debug))
+                    command.LogCommand(connector);
```

Comment:
I'm not sure `LogCommand` works correctly in this case (for example, it starts a `QueryLogStopWatch`)